### PR TITLE
AR shortcode and selector

### PR DESCRIPTION
**Pull request by Camwyn**
🎫 https://central.tri.be/issues/123044

Requires https://github.com/moderntribe/event-tickets-plus/pull/743

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | The method parameter $params is never used | [Shortcode.php:29](https://github.com/moderntribe/event-tickets/blob/44e42acd2654b9dc0aaaff07801ef5f11c20b5a5/src/Tribe/Attendee_Registration/Shortcode.php#L29)
:warning: | The method parameter $shortcode_name is never used | [Shortcode.php:29](https://github.com/moderntribe/event-tickets/blob/44e42acd2654b9dc0aaaff07801ef5f11c20b5a5/src/Tribe/Attendee_Registration/Shortcode.php#L29)
:x: | There must be a single space after the first parenthesis | [Shortcode.php:30](https://github.com/moderntribe/event-tickets/blob/44e42acd2654b9dc0aaaff07801ef5f11c20b5a5/src/Tribe/Attendee_Registration/Shortcode.php#L30)
:x: | There must be a single space before the last parenthesis | [Shortcode.php:30](https://github.com/moderntribe/event-tickets/blob/44e42acd2654b9dc0aaaff07801ef5f11c20b5a5/src/Tribe/Attendee_Registration/Shortcode.php#L30)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Shortcode.php:33](https://github.com/moderntribe/event-tickets/blob/44e42acd2654b9dc0aaaff07801ef5f11c20b5a5/src/Tribe/Attendee_Registration/Shortcode.php#L33)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'true' | [Shortcode.php:33](https://github.com/moderntribe/event-tickets/blob/44e42acd2654b9dc0aaaff07801ef5f11c20b5a5/src/Tribe/Attendee_Registration/Shortcode.php#L33)
:x: | There must be a single space after the first parenthesis | [View.php:28](https://github.com/moderntribe/event-tickets/blob/44e42acd2654b9dc0aaaff07801ef5f11c20b5a5/src/Tribe/Attendee_Registration/View.php#L28)
:x: | There must be a single space before the last parenthesis | [View.php:28](https://github.com/moderntribe/event-tickets/blob/44e42acd2654b9dc0aaaff07801ef5f11c20b5a5/src/Tribe/Attendee_Registration/View.php#L28)
:x: | There must be a single space after the first parenthesis | [View.php:33](https://github.com/moderntribe/event-tickets/blob/44e42acd2654b9dc0aaaff07801ef5f11c20b5a5/src/Tribe/Attendee_Registration/View.php#L33)
:x: | There must be a single space before the last parenthesis | [View.php:33](https://github.com/moderntribe/event-tickets/blob/44e42acd2654b9dc0aaaff07801ef5f11c20b5a5/src/Tribe/Attendee_Registration/View.php#L33)

> via jenkins/codesniffer-phpcs

**Comment by paulmskim** [comment]
Curious, is there a reason why we have a `.env` file committed to our repo?

**Comment by Camwyn** [comment]
The `.env` is for tests in Travis, etc - as opposed to the `.env.local` which should be for your local environment

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space after &quot;=&quot;; 2 found | [Main.php:32](https://github.com/moderntribe/event-tickets/blob/276357404bb5d16eb90291173d8ea2694f6ac65d/src/Tribe/Attendee_Registration/Main.php#L32)
:x: | Expected 1 space after &quot;=&quot;; 2 found | [Main.php:34](https://github.com/moderntribe/event-tickets/blob/276357404bb5d16eb90291173d8ea2694f6ac65d/src/Tribe/Attendee_Registration/Main.php#L34)
:warning: | The method parameter $params is never used | [Shortcode.php:29](https://github.com/moderntribe/event-tickets/blob/276357404bb5d16eb90291173d8ea2694f6ac65d/src/Tribe/Attendee_Registration/Shortcode.php#L29)
:warning: | The method parameter $shortcode_name is never used | [Shortcode.php:29](https://github.com/moderntribe/event-tickets/blob/276357404bb5d16eb90291173d8ea2694f6ac65d/src/Tribe/Attendee_Registration/Shortcode.php#L29)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Shortcode.php:32](https://github.com/moderntribe/event-tickets/blob/276357404bb5d16eb90291173d8ea2694f6ac65d/src/Tribe/Attendee_Registration/Shortcode.php#L32)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'true' | [Shortcode.php:32](https://github.com/moderntribe/event-tickets/blob/276357404bb5d16eb90291173d8ea2694f6ac65d/src/Tribe/Attendee_Registration/Shortcode.php#L32)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected 1 space after &quot;=&quot;; 2 found | [Main.php:32](https://github.com/moderntribe/event-tickets/blob/b4bd0f2de6427ba4ccb42545d863edb2c2d3c17e/src/Tribe/Attendee_Registration/Main.php#L32)
:x: | Expected 1 space after &quot;=&quot;; 2 found | [Main.php:34](https://github.com/moderntribe/event-tickets/blob/b4bd0f2de6427ba4ccb42545d863edb2c2d3c17e/src/Tribe/Attendee_Registration/Main.php#L34)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [Shortcode.php:29](https://github.com/moderntribe/event-tickets/blob/b4bd0f2de6427ba4ccb42545d863edb2c2d3c17e/src/Tribe/Attendee_Registration/Shortcode.php#L29)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not 'true' | [Shortcode.php:29](https://github.com/moderntribe/event-tickets/blob/b4bd0f2de6427ba4ccb42545d863edb2c2d3c17e/src/Tribe/Attendee_Registration/Shortcode.php#L29)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Attendee_Registration/View.php [hunk 1]
```diff
@@ -23,14 +23,13 @@ public function __construct() {
 	 * @param string $content The original page|post content
 	 * @return srting $template The resulting template content
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/View.php]:
```suggestion
	 * @return string The resulting template content
```

File: src/Tribe/Attendee_Registration/View.php [hunk 2]
```diff
@@ -23,14 +23,13 @@ public function __construct() {
 	 * @param string $content The original page|post content
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/View.php]:
```suggestion
	 * @param string $content The original page|post content
	 * @param string $context The context of the rendering
	 *
```

File: src/Tribe/Attendee_Registration/View.php [hunk 3]
```diff
@@ -23,14 +23,13 @@ public function __construct() {
...
 	 * @return srting $template The resulting template content
 	 */
-	public function display_attendee_registration_page( $content = '' ) {
+	public function display_attendee_registration_page( $content = '', $shortcode = false ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/View.php]:
```suggestion
	public function display_attendee_registration_page( $content = '', $context = 'default' ) {
```

File: src/Tribe/Attendee_Registration/View.php [hunk 4]
```diff
@@ -23,14 +23,13 @@ public function __construct() {
...
 		global $wp_query;
 
 		// Bail if we don't have the flag to be in the registration page
-		if ( ! tribe( 'tickets.attendee_registration' )->is_on_page() ) {
+		if ( ! $shortcode && ! tribe( 'tickets.attendee_registration' )->is_on_page() ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/View.php]:
```suggestion
		if ( 'shortcode' === $context && ! tribe( 'tickets.attendee_registration' )->is_on_page() ) {
```

File: src/Tribe/Attendee_Registration/Shortcode.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	 * Renders the shortcode AR page.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function render() {
+		ob_start();
+
+		tribe( 'tickets.attendee_registration.view' )->display_attendee_registration_page( null, true );
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Shortcode.php]:
```suggestion
		tribe( 'tickets.attendee_registration.view' )->display_attendee_registration_page( null, 'shortcode' );
```

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space after the first parenthesis | [View.php:31](https://github.com/moderntribe/event-tickets/blob/f560d323334d691e76a784d1c1cdc8dcff99b10a/src/Tribe/Attendee_Registration/View.php#L31)
:x: | There must be a single space before the last parenthesis | [View.php:31](https://github.com/moderntribe/event-tickets/blob/f560d323334d691e76a784d1c1cdc8dcff99b10a/src/Tribe/Attendee_Registration/View.php#L31)

> via jenkins/codesniffer-phpcs

**Comment by Camwyn** [comment]
Put on holds because I discovered a small issue I'd like to fix after lunch - setting a page that does not contain the shortcode results in errors. We can fallback to the default redirect settings in that case.

File: src/Tribe/Attendee_Registration/View.php [hunk 6]
```diff
@@ -21,16 +21,16 @@ public function __construct() {
 	 *
 	 * @since 4.9
 	 * @param string $content The original page|post content
-	 * @return srting $template The resulting template content
+	 * @param string $context The context of the rendering
+	 * @return string $template The resulting template content
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/View.php]: `@return` doesn't get a $varname

```suggestion
	 * @return string The resulting template content
```

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space before the last parenthesis | [Main.php:36](https://github.com/moderntribe/event-tickets/blob/00c0594a18175aa8124cef600f6f917fc8a5e2db/src/Tribe/Attendee_Registration/Main.php#L36)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Attendee_Registration/Main.php [hunk 7]
```diff
@@ -29,7 +29,15 @@ class Tribe__Tickets__Attendee_Registration__Main {
 	 * @return string
 	 */
 	public function get_slug() {
-		return Tribe__Settings_Manager::get_option( 'ticket-attendee-info-slug', $this->default_page_slug );
+		$slug = Tribe__Settings_Manager::get_option( 'ticket-attendee-page-slug', false );
+
+		$page = get_page_by_path( $slug );
+
+		if ( empty( $slug ) || ! has_shortcode( $page->post_content, 'tribe_attendee_registration') ) {
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/Main.php]:
```suggestion
		if ( empty( $slug ) || ! has_shortcode( $page->post_content, 'tribe_attendee_registration' ) ) {
```

File: src/Tribe/Attendee_Registration/View.php [hunk 8]
```diff
@@ -21,16 +21,16 @@ public function __construct() {
 	 *
 	 * @since 4.9
 	 * @param string $content The original page|post content
-	 * @return srting $template The resulting template content
+	 * @param string $context The context of the rendering
+	 * @return string The resulting template content
```
Comment (sc0ttkclark) [inline src/Tribe/Attendee_Registration/View.php]:
```suggestion
	 *
	 * @return string The resulting template content
```

**Pull request merged**